app/services/open_access_urls_migration_service.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+class OpenAccessUrlsMigrationService
+  def self.call
+    Publication.all.each do |pub|
```

Comment:
The `:all` method is like an anti-pattern, it seemed like a good idea at the time but it doesn't perform well. The method loads up all the records into memory. Use `:find_each` instead.